src/Blockcore.Indexer.Core/Sync/SyncTasks/BlockPuller.cs

```diff
@@ -231,11 +231,13 @@ private void ProcessFromBlockingCollection()
          }
       }
 
-      private static Task<string> NextHashAsync(IBlockchainClient client, long height)
+      private static async Task<string> NextHashAsync(IBlockchainClient client, long height)
       {
          try
          {
-            return client.GetblockHashAsync(height);
+            string blockHash = await client.GetblockHashAsync(height);
```

Comment:
This could be simplified without await and just return.

Reply:
you mean like it was before? there is a funny issue, if we return here then the try/catch doesn't trigger (it will throw outside this method, I think the task only when its awaited (or the result is read) will throw so awaiting here will trigger the exception inside the try/catch.